src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http3.cs

```diff
@@ -25,8 +26,143 @@ public HttpClientHandlerTest_Http3(ITestOutputHelper output) : base(output)
+
+                // Unlike HTTP/2, we can't count on stream processing order.
+                // We must wait a bit to ensure the client has received the SETTINGS frame.
+                await Task.Delay(5_000);
+
+                QuicConnectionAbortedException ex = await Assert.ThrowsAsync<QuicConnectionAbortedException>(async () => await connection.HandleRequestAsync());
+                Assert.Equal(SettingsErrorCode, ex.ErrorCode);
+            });
+
+            Task clientTask = Task.Run(async () =>
+            {
+                using HttpClient client = CreateHttpClient();
+                using HttpRequestMessage request = new()
+                {
+                    Method = HttpMethod.Get,
+                    RequestUri = server.Address,
+                    Version = HttpVersion30,
+                    VersionPolicy = HttpVersionPolicy.RequestVersionExact
+                };
+
+                await Assert.ThrowsAsync<HttpRequestException>(async () => await client.SendAsync(request));
+            });
+
+            await new[] { clientTask, serverTask }.WhenAllOrAnyFailed(20_000);
+        }
+
+        [Fact]
+        public async Task ReservedFrameType_Throws()
+        {
+            const int ReservedHttp2PriorityFrameId = 0x2;
+            const long UnexpectedFrameErrorCode = 0x105;
+
+            using Http3LoopbackServer server = CreateHttp3LoopbackServer();
+
+            Task serverTask = Task.Run(async () =>
+            {
+                using Http3LoopbackConnection connection = (Http3LoopbackConnection)await server.EstablishGenericConnectionAsync();
+                using Http3LoopbackStream stream = await connection.AcceptRequestStreamAsync();
+
+                await stream.SendFrameAsync(ReservedHttp2PriorityFrameId, new byte[8]);
+
+                QuicConnectionAbortedException ex = await Assert.ThrowsAsync<QuicConnectionAbortedException>(async () =>
+                {
+                    await stream.HandleRequestAsync();
+                    using Http3LoopbackStream stream2 = await connection.AcceptRequestStreamAsync();
+                });
+
+                Assert.Equal(UnexpectedFrameErrorCode, ex.ErrorCode);
+            });
+
+            Task clientTask = Task.Run(async () =>
+            {
+                using HttpClient client = CreateHttpClient();
+                using HttpRequestMessage request = new()
+                {
+                    Method = HttpMethod.Get,
+                    RequestUri = server.Address,
+                    Version = HttpVersion30,
+                    VersionPolicy = HttpVersionPolicy.RequestVersionExact
+                };
+
+                await Assert.ThrowsAsync<HttpRequestException>(async () => await client.SendAsync(request));
+            });
+
+            await new[] { clientTask, serverTask }.WhenAllOrAnyFailed(20_000);
+        }
+
+        [OuterLoop]
+        [ConditionalTheory(nameof(IsMsQuicSupported))]
```

Comment:
I wonder if we should have a separate test class specifically for interop. Not needed for this PR of course.